src-electron/db/query-attribute.js

```diff
@@ -32,9 +33,35 @@ function attributeExportMapping(x) {
     type: x.TYPE,
     define: x.DEFINE,
     mfgCode: x.MANUFACTURER_CODE,
+    manufacturerCode: x.MANUFACTURER_CODE,
+    hexCode: '0x' + bin.int16ToHex(x['CODE']),
     clusterSide: x.SIDE,
     clusterName: x.CLUSTER_NAME,
     isClusterEnabled: x.ENABLED,
+    isArray: x.IS_ARRAY,
+    isWritable: dbApi.fromDbBool(x.IS_WRITABLE),
+    isWritableAttribute: dbApi.fromDbBool(x.IS_WRITABLE),
+    isReportableAttribute: dbApi.fromDbBool(x.INCLUDED_REPORTABLE),
+    includedReportable: dbApi.fromDbBool(x.INCLUDED_REPORTABLE),
+    isNullable: dbApi.fromDbBool(x.IS_NULLABLE),
+    maxLength: x.MAX_LENGTH,
+    minLength: x.MIN_LENGTH,
+    arrayType: x.ARRAY_TYPE,
+    entryType: x.ARRAY_TYPE,
+    min: x.MIN,
+    max: x.MAX,
+    mustUseTimedWrite: dbApi.fromDbBool(x.MUST_USE_TIMED_WRITE),
+    isSceneRequired: dbApi.fromDbBool(x.IS_SCENE_REQUIRED),
+    isOptionalAttribute: dbApi.fromDbBool(x.IS_OPTIONAL),
```

Comment:
This is not the name IS_OPTIONAL is exposed with for other helpers....  Every single other mapping uses `isOptional` for this.  In general, why are we not just using or existing dbmapping bits here instead of coming up with a new mapping?

Reply:
The query was using this map so I just extended this one

Reply:
I understand yes. My point is that it led to this map not looking like other maps, which raises the question of why we are not using the other map to start with.

Reply:
I can look into converging this with the other map. Probably some divergence from old code.

Reply:
Thank you.  Followup is OK for that, in general, but for now we should just make sure they match exactly for the new things being added.